include/Buffer.hpp

```diff
@@ -55,7 +55,7 @@ class Buffer {
 
 	inline iterator begin() { return _sent.begin(); }
 	inline iterator end()   { return _sent.end(); }
-	inline size_t size() { return _sent.size(); }
+	inline size_t size() { lock(); size_t s = _sent.size(); unlock(); return s; }
```

Comment:
That's interesting. What I don't like is that this needs to nested lock() calls but it shouldn't really matter. I just wonder whether begin()/end() need a protection as well. In any case begin()/end() wouldn't be sufficient but may only be called from within a lock. So I wonder whether a 2nd public: "size()" (which is mainly used as contains_data()?) with a lock would be a cleaner solution -> all public interfaces are thread safe, only private interface are not but here the Buffer class itself needs to ensure that locking is properly done. Anyhow this change itself is fine for me.